src/inventory_ui.cpp

```diff
@@ -910,7 +915,21 @@ item_location game::inv_map_splice(
                     }
                 }
             }
-            inv_s.make_item_list( slices.back(), pos == g->u.pos() ? &ground_cat : &nearby_cat );
```

Comment:
If no filtered item is found in the loop, there is no need to call `inv_s.make_item_list`.